Core/GDCore/Project/ObjectGroupsContainer.cpp

```diff
@@ -4,101 +4,147 @@
+  if (i != objectGroups.end()) return **i;
 
   return badGroup;
 }
 
 const ObjectGroup& ObjectGroupsContainer::Get(const gd::String& name) const {
-  auto i = std::find_if(
-      objectGroups.begin(),
-      objectGroups.end(),
-      [&name](const ObjectGroup& group) { return group.GetName() == name; });
-  if (i != objectGroups.end()) return *i;
+  auto i = std::find_if(objectGroups.begin(),
+                        objectGroups.end(),
+                        [&name](const std::unique_ptr<gd::ObjectGroup>& group) {
+                          return group->GetName() == name;
+                        });
+  if (i != objectGroups.end()) return **i;
 
   return badGroup;
 }
 
-ObjectGroup& ObjectGroupsContainer::Insert(const gd::ObjectGroup& group,
-                                           std::size_t position) {
-  if (position < objectGroups.size()) {
-    objectGroups.insert(objectGroups.begin() + position, group);
-    return objectGroups[position];
-  } else {
-    objectGroups.push_back(group);
-    return objectGroups.back();
-  }
-}
-
-#if defined(GD_IDE_ONLY)
 void ObjectGroupsContainer::Remove(const gd::String& name) {
-  objectGroups.erase(std::remove_if(objectGroups.begin(),
-                                    objectGroups.end(),
-                                    [&name](const ObjectGroup& group) {
-                                      return group.GetName() == name;
-                                    }),
-                     objectGroups.end());
+  objectGroups.erase(
+      std::remove_if(objectGroups.begin(),
+                     objectGroups.end(),
+                     [&name](const std::unique_ptr<gd::ObjectGroup>& group) {
+                       return group->GetName() == name;
+                     }),
+      objectGroups.end());
 }
 
 std::size_t ObjectGroupsContainer::GetPosition(const gd::String& name) const {
   for (std::size_t i = 0; i < objectGroups.size(); ++i) {
-    if (objectGroups[i].GetName() == name) return i;
+    if (objectGroups[i]->GetName() == name) return i;
   }
 
   return gd::String::npos;
 }
 
 ObjectGroup& ObjectGroupsContainer::InsertNew(const gd::String& name,
                                               std::size_t position) {
-  ObjectGroup newGroup;
-  newGroup.SetName(name);
-  return Insert(newGroup, position);
+  gd::ObjectGroup& newlyInsertedGroup = *(*(objectGroups.insert(
+      position < objectGroups.size() ? objectGroups.begin() + position
+                                     : objectGroups.end(),
+      gd::make_unique<gd::ObjectGroup>())));
+  newlyInsertedGroup.SetName(name);
+  return newlyInsertedGroup;
+}
+
+ObjectGroup& ObjectGroupsContainer::Insert(const gd::ObjectGroup& group,
+                                           std::size_t position) {
+  gd::ObjectGroup& newlyInsertedGroup = *(*(objectGroups.insert(
+      position < objectGroups.size() ? objectGroups.begin() + position
+                                     : objectGroups.end(),
+      group.Clone())));
```

Comment:
Replace by `gd::make_unique<gd::ObjectGroup>(group)` (which will call the default copy constructor, and will return a unique_ptr to it).